pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2655,6 +2655,18 @@ def test_mergeerror_on_left_index_mismatched_dtypes():
         merge(df_1, df_2, on=["C"], left_index=True)
 
 
+def test_merge_on_left_categoricalindex():
+    # GH#48464 don't raise when left_on is a CategoricalIndex
+    ci = CategoricalIndex(range(3))
+
+    right = DataFrame({"A": ci, "B": range(3)})
+    left = DataFrame({"C": range(3, 6)})
+
+    res = merge(left, right, left_on=ci, right_on="A")
```

Comment:
Should the comment below the code change:

```
           # if either left or right is a categorical
            # then the must match exactly in categories & ordered
```

Apply here as well? Could you add a test where categories/ordered are different too?

Reply:
really shouldn't be relevant.  its just a matter of CategoricalIndex vs Categorical